api/src/rest/permissions.py

```diff
@@ -8,5 +8,4 @@ class IsPetPermission(permissions.BasePermission):
     message = 'Bad PETNAME. No PET associated with this name.'
 
     def has_permission(self, request, view):
-        print('PERMISSION', request.entity)
         return request.entity is not None
```

Comment:
Czy to nie powinno być bardziej zaawanosowane. Jeśli dobrze rozumiem teraz każdy z ważnym certem i wpisem w bazie może przeglądać pozostałe Entity, tak?

Reply:
Przy zapytaniu z ważnym certem klienta, middleware w api/src/rest/middleware.py skojarzy dany request z obiektem `Entity`. Dalej, widoki/endpointy, które mają być dostępne dla klienta sprawdzają czy taki obiekt entity jest przypisany do request-a (korzystając właśnie z `IsPetPermission`) i na jego podstawie zwracają dane. Więc przykładowo zapytanie skierowane na endpoint `/api/entity` zwróci dane tylko dla entity identyfikowanego przez dany certyfikat.

Reply:
Nadal tego nie widzę.

To co robi middleware.py to weryfikuje czy CN jest przypisany do jakiegoś entity w bazie. I bardzo dobrze to uniemożliwi wykorzystanie dowolnego certyfikatu wystawionego przez nasze CA.

W permissions.py sprawdzasz jedynie czy takie entity zostało znalezione. Nie widzę nigdzie testu czy request.entity jest zgodne z tym czego dotyczy zapytanie.

Obecnie wydaje mi się że entity A jest w stanie odpytać API o wszystkie entity które są w bazie.

Reply:
Myślałem nad rozdzieleniem klientów od adminów na podstawie dodatkowego pola w certyfikacie (jeśli tak się da). Ewentualnie skorzystać z djangowego modelu user-a. Część endpointów (te generalne, np listujące wszystkie Entity) ograniczyć dla adminów, a część (np proste http://root-url/entity) które mają sens w kontekście konkretnego entity (certu) ograniczyć dla userów.